test/api/v3/integration/groups/POST-groups_groupId_join.js

```diff
@@ -0,0 +1,63 @@
+import {
+  generateUser,
+} from '../../../../helpers/api-integration.helper';
+
+describe('POST /group/:groupId/join', () => {
+  let user;
+
+  beforeEach(() => {
+    return generateUser().then((generatedUser) => {
+      user = generatedUser;
+    });
+  });
+
+  context('Guilds', () => {
+    it('inviting to guild doesn\'t give basilist to inviter', () => {
+      let groupName = 'Test Guild';
+      let groupType = 'guild';
+
+      return generateUser({balance: 1, 'items.quests.basilist': 0 }).then((generatedUser) => {
+        user = generatedUser;
+        return user.post('/groups', {
+          name: groupName,
+          type: groupType
+        });
+      })
+      .then((guild) => {
+        return generateUser({ 'invitations.guilds': [{ id: guild._id}]}).then((generatedUser) => {
+          return generatedUser.post('/groups/' + guild._id + '/join');

```

Comment:
While writing it I changed my opinion on what's better like 5 times. On one hand calling the API is better, but sometimes you need to set a lot of other stuff to make an API call work, that will just set a variable for you. It slows down the unit test, but also makes it resilient to change in the implementation of said API route. If we don't care about unit tests running time and the API call can be executed without a lot of hassle, I think it's the better choice. If it's easier to mock the API route, than to set up the API call to actually run, maybe mocking is better.